drivers: serial: native_posix: Convert 2nd UART to be devicetree based

galak opened the pull request:
The first uart instance was already devicetree based.  To be consistent
convert the second instance to also be devicetree based.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

daor-oti commented on boards/posix/native_posix/doc/index.rst:
```diff
@@ -587,8 +587,6 @@ UART
 to instantiate up to two UARTs. By default only one UART is enabled.
-With :option:`CONFIG_UART_NATIVE_POSIX_PORT_1_ENABLE`
-you can enable the second one.
 
```
As there is now no configure option to enable the second uart, do we not need to describe how to get the second uart enabled?
The native_posix.dts only defines one uart, so perhaps we would need to explain the process of hoe to get the second uart added?

galak replied:
We can add the second serial port to the DTS and mark it 'disabled'.

daor-oti replied:
That is a possibility, but could we not leave the UART_NATIVE_POSIX_PORT_1_ENABLE Kconfig option to enable uart 1? Then we could also leave the depends on for the Kconfig option UART_NATIVE_WAIT_PTS_READY_ENABLE. Or perhaps it is not possible for a Kconfig option to set a DT status option to okay?

galak replied:
Its not possible.  We parse DTS before Kconfig, so we can't use any Kconfig to set DTS info.

daor-oti commented on drivers/serial/Kconfig.native_posix:
```diff
@@ -41,29 +41,14 @@ endchoice
 config UART_NATIVE_WAIT_PTS_READY_ENABLE
 	bool "Support waiting for pseudo terminal client readiness"
-	depends on NATIVE_UART_0_ON_OWN_PTY || UART_NATIVE_POSIX_PORT_1_ENABLE
+	depends on !NATIVE_UART_0_ON_STDINOUT
```
This can now not be enabled for uart 1, when uart 0 is on stdin/out.

galak replied:
ah, yeah.. the logic didn't work out here.  How do we feel about removing the `depends` on?

daor-oti reviewed (changes requested):
In general looks good, just a few concerns.

aescolar reviewed (dismissed):
I'm not incredible keen on this PR as it is:
* The benefit of the change is being more consistent using DTS for the 2nd UART
* The drawback is a incompatible change for users, it being more difficult to enable the 2nd UART, and needing to drop that kconfig dependency.

Would it be an option to leave the enabling of the UART as a kconfig option, and just move the device macro and name to DTS? (That way we avoid the drawbacks).

galak commented:
> I'm not incredible keen on this PR as it is:
> 
> * The benefit of the change is being more consistent using DTS for the 2nd UART
> * The drawback is a incompatible change for users, it being more difficult to enable the 2nd UART, and needing to drop that kconfig dependency.
> 
> Would it be an option to leave the enabling of the UART as a kconfig option, and just move the device macro and name to DTS? (That way we avoid the drawbacks).

Let me take a look, but i think that might be feasible.

galak committed:
drivers: serial: native_posix: Convert 2nd UART to be devicetree based

The first uart instance was already devicetree based.  To be consistent
convert the second instance to also be devicetree based.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@daor-oti can you take another look now.
